tests/test_jshandle.py

```diff
@@ -0,0 +1,212 @@
+import pytest
+from syncer import sync
+
+from pyppeteer.errors import NetworkError
+from tests.conftest import CHROME
+
+
+@sync
+async def test_handle(isolated_page, server):
+    """test getting handle"""
+    p = isolated_page
+    assert await p.evaluateHandle('window')
+
+
+@sync
+@pytest.mark.skip('NotImplemented - evaluate functions do not support obj references')
+async def test_handle_with_arg(isolated_page, server):
+    """test getting handle"""
+    p = isolated_page
+    navigatorHandle = await p.evaluateHandle("navigator")
+    # this should be able to take local reference
+    text = await p.evaluate('e => e.userAgent', navigatorHandle)
+    assert 'Mozilla' in text
+
+
+@sync
+async def test_handle_primitive_types(isolated_page, server):
+    """test getting handle"""
+    p = isolated_page
+    handle = await p.evaluateHandle('5')
+    assert p.evaluate('e => Object.is(e, 5)', handle)
+
+
+@sync
+async def test_warn_nested_handles(isolated_page, server):
+    p = isolated_page
+    handle = await p.evaluateHandle('document.body')
+    await p.evaluateHandle(
+        "opts => opts.elem.querySelector('p')",
+        handle,  # todo translate {elem: handle}
```

Comment:
It's JS's object literal notation:
```suggestion
        {'opts': handle}
```

Reply:
Unfortunately this test will hang and get terminated by `pytest-timeout`

Reply:
yeah that does hang the whole test suit, that's why I left it as a comment. Do you know what causes the hang? I think some some signals are malformed.

Reply:
Not sure why it hangs. I'm guessing we're getting a bit of a different response from the browser and not handling it correctly, leading to a send future that's never resolved. I've mitigated the effects of this by reducing the test timeout from 120s to just 10s. Our longest test in the suite (so far) takes around 5s on my dual core laptop from years ago, so I figured 10s just to be safe.